src/Handlers.cpp.Rt

```diff
@@ -916,10 +918,15 @@ class cbFailcheck: public Callback {
 		}
 		<?R }; ifdef(); ?>
 		MPI_Allreduce(&cond,&fin,1,MPI_INT,MPI_LOR,MPI_COMM_WORLD);
-		if (fin) { 
-			 notice("Stopping due to Nan value\n");
-			 ret = ITERATION_STOP;
-		}
+	        if ((fin) && (rkept)) {
+                         rkept = 0;
```

Comment:
Make a "notice" also here. something like "Nan value discovered. Executing the final wishes from the FailCheck element." :-)